core/app/api/customer/groups/route.ts

```diff
@@ -0,0 +1,38 @@
+import { NextResponse } from 'next/server';
+import { z } from 'zod';
+
+export const CustomersGroup = z
+  .object({
+    id: z.number(),
+    name: z.string(),
+  })
+  .array();
+
+export type CustomerGroupType = z.infer<typeof CustomersGroup>;
+
+export async function GET() {
+  if (!process.env.BIGCOMMERCE_ACCESS_TOKEN) {
+    throw new Error('BIGCOMMERCE_ACCESS_TOKEN is not set');
+  }
+
+  const response = await fetch(
+    `https://api.bigcommerce.com/stores/${process.env.BIGCOMMERCE_STORE_HASH}/v2/customer_groups`,
+    {
+      method: 'GET',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+        'X-Auth-Token': process.env.BIGCOMMERCE_ACCESS_TOKEN,
+      },
+    },
+  );
+
+  if (!response.ok) {
```

Comment:
`fetch` can still throw an error which isn't covered here. `response.ok` only checks for a status between 200-300. I would wrap this whole function in a try/catch. Then we can check for different instances of errors and return appropriate data. Looking at `core/app/api/products/group/[group]/route.ts`, here's the general error handling there.

```
  if (error instanceof Error) {
      return { status: 'error', error: error.message };
    }

    return { status: 'error', error: 'Something went wrong. Please try again.' };
```